Source/JavaScriptCore/yarr/YarrParser.h

```diff
@@ -763,7 +763,7 @@ class Parser {
             switch (peek()) {
             case '|':
                 consume();
-                m_delegate.disjunction();
+                m_delegate.disjunction(true);
```

Comment:
Can you use something other than simply `true`?  The best practice for cases like this is either to use an enum class, or at least declare a named variable here (e.g. `constexpr bool hasNextAlternative = true;`), and use that named variable instead.

Also, down below, you named the parameter `nextAlternative`, but you're not really passing a "next alternative" here.  Instead, you're making some statement.  Is it "hasNextAlernative", or something else?  Can you rename the parameter to be clearer about it's meaning?

Reply:
We are creating the next alternative in this case.  The value is either, we are creating a "next" alternative or we are not crating a "next" alternative.  I'll change to an enum.